logging_config.yaml

```diff
@@ -0,0 +1,37 @@
+# It contains the configuration for logging
+# You can find more info in https://docs.python.org/3/library/logging.config.html#logging-config-api
+version: 1
+formatters:
+  detailed:
+    format: "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+filters:
+  warnings_filter:
+    (): logging.Filter
+    name: "py.warnings"
+  info_only_filter:
+    (): GANDLF.utils.gandlf_logger.InfoOnlyFilter
+handlers:
+  stdoutHandler: # only display info level
+    class: logging.StreamHandler
+    level: INFO
+    formatter: detailed
+    filters: [info_only_filter]
+    stream: ext://sys.stdout
+  stderrHandler: # display warning and above messages
+    class: logging.StreamHandler
+    level: ERROR
+    formatter: detailed
+    stream: ext://sys.stderr
+  debugHandler:
+    class: logging.FileHandler
+    filename: gandlf.log
+    formatter: detailed
+    level: DEBUG
```

Comment:
Now `gandlf.log` is always appended, mixing logs from different runs. Is it ok? IMO after 10+ commands it would be hard to find any useful info there. Should we rewrite the file on every command? @benmalef @sarthakpati

Reply:
Yes, it makes sense to me. 
Can we use the [rotating file handler](https://docs.python.org/3/library/logging.handlers.html#rotatingfilehandler) or simply rewrite the file ? @VukW

Reply:
As we discussed with @sarthakpati here https://github.com/mlcommons/GaNDLF/pull/842#discussion_r1570850957 , lets make it rotating file handler somewhere in `/tmp/gandlf/` ?